### PR TITLE
MM-49162: Fix Pinned RHS not loading in Threads view

### PR DESCRIPTION
**Pull request by calebroseland**
#### Summary
Simplifies `isRHSLoading` conditions to their respective roots. This continues to solve the original issue: `white screen crash when coming back to channels from system console when RhsThread, ChannelInfoRhs, etc. are open `, but with simplified granularity on specific dependencies—now inline with the `content` assignment blocks.

#### Ticket Link
- Fixes https://mattermost.atlassian.net/browse/MM-49162

#### Screenshots
CleanShot 2022-12-15 at 16 32 35

#### Release Note

```release-note
Fixed an issue where clicking Pinned on a post in Threads view would result in RHS being stuck in a loading state.
```

### CODE REVIEW

**Comment by calebroseland** [comment]
Note: Still need to test/verify a few flows

**Comment by michelengelen** [comment]
@calebroseland ping me once you are ready with this! :D

**Comment by calebroseland** [comment]
@michelengelen, @M-ZubairAhmed, back to ready on this 👍

File: components/sidebar_right/sidebar_right.tsx [hunk 1]
```diff
@@ -230,19 +233,18 @@ export default class SidebarRight extends React.PureComponent<Props, State> {
...
-            );
+            currentChannelNeeded = true;
+            content = <ChannelMembersRhs/>;
         }
 
-        // Sometimes the channel/team is not loaded yet, so we need to wait for it
-        const isChannelSpecificRHS = postRightVisible || postCardVisible || isPinnedPosts || isChannelFiles || isChannelInfo || isChannelMembers;
-
-        const isRHSLoading = postRightVisible ? !(team || channel) : (!team || !channel) && isChannelSpecificRHS;
+        const isRHSLoading = Boolean(
```
Comment (michelengelen) [inline components/sidebar_right/sidebar_right.tsx:243]: nit: I think the extra `Boolean` is not needed

**Review by michelengelen — APPROVED** [review]
lgtm ... just one very small nit

**Review by jgilliam17 — APPROVED** [review]
Thanks @calebroseland 
Tested, looks good to merge.
Verified pinned, saved, search and mention rhs while in threads view. Additionally, checked for any issues when returning from sys console to channels with rhs open - working as expected.
[E2E report](https://automation-dashboard.vercel.app/cycle/284152) - no PR related failures.

**Comment by jgilliam17** [comment]
@amyblais Which version should I mark this ticket for? Is it next cloud 01/19/23?

**Comment by amyblais** [comment]
> Which version should I mark this ticket for? Is it next cloud 01/19/23?

@jgilliam17 That's accurate.

**Pull request merged**